.github/workflows/setup_env.sh

```diff
@@ -105,6 +105,13 @@ quiet which mpif90
 mpicxx --version
 mpif90 --version
 
+#----------------------------------------------------------------- GPU
+# Find idle GPUs. gpu_bind.sh also uses these.
+export idle_gpus=$(./tools/idle_gpus.py)
+export idle_gpus=(${idle_gpus})       # convert to array
+export gpu_kind=${idle_gpus[0]}       # element 0 is gpu_kind: cuda or rocm
```

Comment:
When you say "pass $device", do you mean something like `./idle_gpus.py 0`, which would return just the 0-th idle GPU? So we could set
```
export idle_gpus=$(./tools/idle_gpus.py 0)
```
I think the `idle_gpus.py` output makes more sense if you look at the `ci_mpi` branch. (There `idle_gpus.py` was named `available_gpus.py`, but I think `idle_gpus.py` is more descriptive.) There, `gpu_bind.sh` needs the `gpu_kind`, and it looks at all the available idle GPUs to decide which to keep visible.

Reply:
`$device` contains whether we're on NVIDIA, AMD, or Intel.  So, I was suggesting using `./idle_gpus.py "${device}"` instead of making `idle_gpus.py` figure out what kind of device we have.